syntaxes/styled-components.json

```diff
@@ -59,6 +59,47 @@
 					"include": "source.css.styled"
 				}
 			]
+		},
+		{
+			"name": "styled.withComponent",
+			"begin": "(?:(?:([_$[:alpha:]][_$[:alnum:]]*)(\\.)(withComponent)(?:(?:\\.([_$[:alpha:]][_$[:alnum:]]*))|(?:\\((['\"][_$[:alpha:]][_$[:alnum:]]*['\"])\\))|(?:\\(([_$[:alpha:]][_$\\.[:alnum:]]*)\\)))?)|(css|keyframes|injectGlobal))\\s{0,1}(`)",
```

Comment:
I think the `|(css|keyframes|injectGlobal)` part is unnecessary here. Removing it will also move the capture 8 to 7.